docs/lib/search/index.ts

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from "react";
+import { mockSearchData } from "../mock-search-data";
+import type {
+  PagefindSearchFragment,
+  PagefindSearchResult,
+  PagefindSearchResults,
+} from "./search-types";
+
+export const ignoredRoutes = ["/blog", "/terms", "/privacy", "/confirm"];
+
+export const downrankedRoutes = [
+  "/repo/docs/acknowledgements",
+  // Deprecations
+  "/repo/docs/core-concepts/scopes",
+];
+
+declare global {
+  interface Window {
+    pagefind?: {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Not sure where this type comes from.
+      search: (...args: any[]) => Promise<PagefindSearchResults>;
+    };
+  }
+}
+
+export const usePageFindSearch = () => {
+  useEffect(() => {
+    async function loadPagefind() {
+      if (typeof window.pagefind === "undefined") {
+        try {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- Not sure where to get this type.
+          window.pagefind = await import(
+            // @ts-expect-error -- Generated at buildtime
+            // eslint-disable-next-line import/no-unresolved -- Generated at buildtime
+            /* webpackIgnore: true */ "./pagefind/pagefind.js"
+          );
```

Comment:
This could probably be handled to work better for devtime but I couldn't quite manage it. If anyone has advice, happy to hear it!